doc/source/build_hints.rst

```diff
@@ -1803,11 +1803,68 @@ The following options are available to select a subset of drivers:
     Independently of options that control global behavior, drivers can be individually
     enabled or disabled with those options.
 
+    .. note::
+
+        <driver_name> above and below is *generally*, but not systematically the short driver name.
+
+        Some drivers may also be grouped together for build purposes.
+
+        - A number of "raw" raster drivers (ACE2, BT, BYN, CPG, CTable2, DIPX, DOQ1,
+          DOQ2, EHDR, EIR, ENVI, FAST, GenBIN, GSC, GTX, MFF2, ISCE, KRO, MFF, LAN,
+          LCP, LOSLAS, NDF, NTv2, PAUX, PNM, ROIPAC, RRASTER, SNODAS) are controlled
+          by the GDAL_ENABLE_DRIVER_RAW option.
+
+        - Planetary raster formats (PDS, PDS4, ISIS, ISIS3, VICAR) are controlled by
+          the GDAL_ENABLE_DRIVER_PDS option.
+
+        - The ECW and JP2ECW raster drivers are controlled by the GDAL_ENABLE_DRIVER_ECW option.
+
+        - The GSAG, GSBG and GS7BG raster drivers are controlled by the GDAL_ENABLE_DRIVER_GSG option.
+
+        - The MrSID and JP2MrSID raster drivers are controlled by the GDAL_ENABLE_DRIVER_MRSID option.
+
+        - The NITF, RPFTOC and ECRGTOC raster drivers are controlled by the GDAL_ENABLE_DRIVER_NITF option.
+
+        - The Interlis 1 and Interlis 2 vector drivers are controlled by the GDAL_ENABLE_DRIVER_ILI option.
+
+        - The WFS and OAPIF vector drivers are controlled by the GDAL_ENABLE_DRIVER_WFS option.
+
+        - The AVCBIN and AVCE00 vector drivers are controlled by the GDAL_ENABLE_DRIVER_AVC option.
+
+        - The DWG and DGNv8 vector drivers are controlled by the GDAL_ENABLE_DRIVER_DWG option.
+
+        There might be variations in naming, e.g. :
+
+        - the "RST" raster driver is controlled by GDAL_ENABLE_DRIVER_IDRISI.
+
+        - the "ElasticSearch" vector driver is controlled by OGR_ENABLE_DRIVER_ELASTIC.
+
+        - the "PostgreSQL" vector driver is controlled by OGR_ENABLE_DRIVER_PG.
+
+        - the "UK .NTF" vector driver is controlled by OGR_ENABLE_DRIVER_NTF.
+
+    .. note::
+
+        Drivers that have both a raster and vector side (and are internally implemented by a
+        single GDALDriver instance) are controlled by either a GDAL_ENABLE_DRIVER_<driver_name>
+        option or a OGR_ENABLE_DRIVER_<driver_name> one, but not both:
+
+        - The netCDF drivers are controlled by the GDAL_ENABLE_DRIVER_NETCDF option.
+        - The PDF drivers are controlled by the GDAL_ENABLE_DRIVER_PDF option.
+        - The GPKG drivers are controlled by the OGR_ENABLE_DRIVER_GPKG option.
+        - The SQLite drivers are controlled by the OGR_ENABLE_DRIVER_SQLITE option.
```

Comment:
Should GRASS be included in this list?

Reply:
Actually, GRASS doesn't have a flag for the raster or vector driver...

Reply:
The GRASS driver has been removed from this repository and moved to a dedicate one as indicated in https://github.com/OSGeo/gdal/blob/v3.5.0/NEWS.md#in-a-nutshell

Reply:
Should it be removed from https://gdal.org/drivers/raster/index.html and https://gdal.org/drivers/vector/index.html then?

Reply:
Does that also include [GRASSASCIIGrid](https://gdal.org/drivers/raster/grassasciigrid.html#raster-grassasciigrid)?

Reply:
> Should it be removed from https://gdal.org/drivers/raster/index.html and https://gdal.org/drivers/vector/index.html then?

no, we want to keep those entries so that users are pointed to the new repository

> Does that also include [GRASSASCIIGrid](https://gdal.org/drivers/raster/grassasciigrid.html#raster-grassasciigrid)?

no, it is controlled by GDAL_ENABLE_DRIVER_AAIGRID